src/docquery/document.py

```diff
@@ -215,25 +156,42 @@ def context(self) -> Tuple[(str, List[int])]:
 
 
 @validate_arguments
-def load_document(fpath: str):
+def load_document(fpath: str, ocr_processor_name=Optional[str]):
     if fpath.startswith("http://") or fpath.startswith("https://"):
         resp = requests.get(fpath, stream=True)
         if not resp.ok:
             raise UnsupportedDocument(f"Failed to download: {resp.content}")
         b = resp.raw
     else:
         b = open(fpath, "rb")
-    return load_bytes(b, fpath)
+    return load_bytes(b, fpath, ocr_processor_name)
 
 
-def load_bytes(b, fpath):
+def load_bytes(b, fpath, ocr_processor_name=Optional[str]):
+    ocr_processor = get_ocr_processor(ocr_processor_name)
     extension = os.path.basename(fpath).rsplit(".", 1)[-1].split("?")[0].strip()
     if extension in ("pdf"):
-        return PDFDocument(b.read())
+        return PDFDocument(b.read(), ocr_processor=ocr_processor)
     else:
         use_pil()
         try:
             img = Image.open(b)
         except UnidentifiedImageError as e:
             raise UnsupportedDocument(e)
-        return ImageDocument(img)
+        return ImageDocument(img, ocr_processor=ocr_processor)
+
+
+def get_ocr_processor(ocr_processor_name: Optional[str]):
+    if not ocr_processor_name:
+        if TESSERACT_AVAILABLE:
+            return TesseractProcessor()
+        elif EASYOCR_AVAILABLE:
+            return EasyOCRProcessor()
+        else:
+            DummyProcessor()
```

Comment:
I think you need a return in front of `DummyProcessor()` here and line 197

Reply:
Lol whops this is why we need some tests. Good catch